[Yshinprograms] iP

Yshinprograms opened the pull request.

Yshinprograms committed:
Add new name & greeting

Yshinprograms committed:
Add command echo function

Yshinprograms committed:
Add ability to store user inputs and display when requested

Yshinprograms committed:
Add ability to mark tasks as done & revert to undone

Yshinprograms committed:
Applied SLAP to existing code

Yshinprograms committed:
Add TaskList class to handle Task operations

Yshinprograms committed:
Add Todo and Deadline classes & fixed "unmark" and "mark" order in Clod class

Yshinprograms committed:
Add Event class & error checking for missing /by for Deadline class

Yshinprograms committed:
Add automated Text UI Testing & Removed Clod logo

jing-yen commented on src/main/java/Deadline.java:
```diff
@@ -0,0 +1,61 @@
+public class Deadline extends Task {
+    private static final String DEADLINE_PREFIX = "deadline";
+    private static final String TYPE_ICON = "D";
+    private static final String BY_DELIMITER = "/by";
+    private String by;
```
Great job using camelCase for variables and SCREAMING_SNAKE_CASE for constants!

jing-yen commented on src/main/java/Event.java:
```diff
@@ -0,0 +1,96 @@
+        }
+
+        String fromAndTo = userInput[1];
+        String[] fromAndToSplit = fromAndTo.split(TO_DELIMITER, 2);
```
Good job attaching array specifiers to the type (String) rather than the variable (fromAndToSplit).

jing-yen commented on src/main/java/TaskList.java:
```diff
@@ -0,0 +1,86 @@
+import java.util.ArrayList;
+
+class TaskList {
+    public static final int LISTLOWERBOUND = 1;
+    public static final String LISTSEPARATOR = "==============================================";
```
I think you are using SCREAMING_SNAKE_CASE as they are constants. Should they be named LIST_LOWER_BOUND and LIST_SEPARATOR?

jing-yen commented on src/main/java/TaskList.java:
```diff
@@ -0,0 +1,86 @@
+            try {
+                return Integer.parseInt(word);
+            } catch (NumberFormatException ignored) {
+            }
```
Nice use of the try-catch block. Perhaps the catch block should have a statement e.g. a print function to notify the user of any exceptions that occurred.

jing-yen commented on src/main/java/Event.java:
```diff
@@ -0,0 +1,96 @@
+        setEnd("");
+    }
+
+    private boolean missingTo(String[] fromAndToSplit) {
```
I didn't like how the missingTo variable was named. For Boolean methods, you can consider making the names sound like booleans e.g. using prefixes of 'is', 'has', 'was'.

jing-yen commented:
A great, comprehensive program with great code quality!

ae-24 commented on src/main/java/Clod.java:
```diff
@@ -0,0 +1,106 @@
+            return "event";
+        }
+        return "add";
+    }
```
Perhaps combine processUserCommand and parseCommand into a single function to reduce duplication of logic and improve maintainability.

ae-24 commented on src/main/java/Task.java:
```diff
@@ -0,0 +1,45 @@
+        isDone = done;
+    }
+
+    public String getDescription() {
+        return "[" + getStatusIcon() + "] " + description;
+    }
+
+    public String getBaseDescription() {
+        return description;
+    }
```
Perhaps rename the methods to getFormattedDescription and getDescription to improve clarity and reduce confusion.

ae-24 commented on src/main/java/Clod.java:
```diff
@@ -0,0 +1,106 @@
+        return in.nextLine();
+    }
+
+    private static void promptUserInputWith(String x) {
+        printUserPrefix(false);
+        System.out.println(x);
+    }
```
Since promptUserInputWith is used only once, perhaps replacing it with a direct System.out.println() call  would simplify the code and improve readability.

ae-24 commented on src/main/java/Clod.java:
```diff
@@ -0,0 +1,106 @@
+            listOfUserInputs.markTaskAsCompleted(taskNumber);
+            break;
+        case "unmark":
+            int taskNumber2 = listOfUserInputs.getUserTaskNumber(userInput);
```
I like how your method and variable names are very descriptive, which is great! However, maybe you might want to consider shortening them slightly to improve readability and keep the code more concise.

Yshinprograms committed:
Improved readability according to github comments

Yshinprograms committed:
Added error handling and initial package structure

This commit introduces basic error and exception handling for user commands (mark, unmark, deadline, todo).

It also sets up the initial package structure:
- clod: Main chatbot class and related logic
- exceptions: Custom exception classes
- operations: Task-related classes (Task, Todo, Deadline, Event, TaskList)

Yshinprograms committed:
Added exception handling for events

Yshinprograms committed:
Merge branch 'branch-Level-5'

Yshinprograms committed:
Added exceptions and operations as packages in the clod package

Yshinprograms committed:
Merge branch 'branch-A-Packages'

Yshinprograms committed:
Combined adding Todo/Deadline/Event into a single function

Yshinprograms committed:
Added support for deleting tasks from the list

Yshinprograms committed:
Added auto saving and loading of data to and from hard disk

- Clod class looks for any previously saved data
- New Storage class creates new data file if no previous data exists
- Storage class automatically converts data to and from string for saving
- TaskList class automatically saves and loads data

Yshinprograms committed:
Merge branch 'branch-level-6'

Yshinprograms committed:
Merge branch 'branch-level-7'

# Conflicts:
#	src/main/java/clod/Clod.java
#	src/main/java/clod/operations/Task.java

Yshinprograms committed:
Refactored code from 'Clod' into 'Interactions' and 'Parser' & fixed save function upon deleting a task

Yshinprograms committed:
Added TimeManager class

Enabled Clod to "understand" time formatting and perform actions on time operations
Changed saving and loading features to parse LocalDateTime members correctly instead of just reading a String

Yshinprograms committed:
Added usage instructions for commands

Yshinprograms committed:
Added find command and function

Yshinprograms committed:
Added JavaDoc comments to the code

Yshinprograms committed:
Merge Pull Request to merge Branch-Level-8 with master branch

Addition of:

TimeManager class
LocalDateTime formatting
Proper handling of Dates and Times
Proper saving and loading of Dates and Times

Yshinprograms committed:
Merge Pull Request to merge Branch-Level-9

Addition of:

- Find command in Parser class
- Find function in TaskList

Yshinprograms committed:
Merge branch 'master' of https://github.com/Yshinprograms/ip

* 'master' of https://github.com/Yshinprograms/ip:
  Added find command and function
  Added usage instructions for commands
  Added TimeManager class

Yshinprograms committed:
Merge branch 'branch-A-JavaDoc'

# Conflicts:
#	src/main/java/clod/operations/Task.java
#	src/main/java/clod/operations/TaskList.java

Yshinprograms committed:
Added User Guide

Yshinprograms committed:
Added README to docs/README.md

Yshinprograms committed:
Added extra JavaDoc comments for TimeManager and Deadline classes

Yshinprograms committed:
Added extra explanations for doc/README.md

Yshinprograms committed:
Added more details to UserGuide and fixed bug in save feature

- Mark and Unmark previously did not save because they did not call saveTaskToStorage()

Yshinprograms committed:
Update README.md on github repo

Yshinprograms committed:
Removed Command Summary from README.md

Yshinprograms committed:
Merge branch 'master' of https://github.com/Yshinprograms/ip

Yshinprograms committed:
Fixed error handling for incorrect DateTime input

okkhoy commented on src/main/java/clod/Clod.java:
```diff
@@ -0,0 +1,55 @@
+    }
+
+    /**
+     * The main execution loop for the application.
```
comment doesn't follow the cs2113 standard

okkhoy commented on src/main/java/clod/Clod.java:
```diff
@@ -0,0 +1,55 @@
+        }
+        scanner.close();
+    }
+}
```
add a new line at the end of the file to prevent github complaining

okkhoy commented on src/main/java/clod/exceptions/ClodException.java:
```diff
@@ -0,0 +1,7 @@
+package clod.exceptions;
+
+public class ClodException extends Exception{
```
need space between `){`

okkhoy commented on src/main/java/clod/operations/TaskList.java:
```diff
@@ -0,0 +1,199 @@
+            System.out.println(LIST_SEPARATOR);
+        }
+    }
+}
```
consider separating out task related methods to a different class rather than putting it under task list

okkhoy commented on src/main/java/clod/operations/TimeManager.java:
```diff
@@ -0,0 +1,68 @@
+
+import clod.exceptions.ClodException;
+
+// LocalDateTime represents yyyy-mm-dd HH:mm
+// LocalDate represents yyyy-mm-dd
+// LocalTime represents HH:mm
+// DateTimeFormatter allows us to define how we want to interpret date/time strings
+// parse() converts a string into date/time
+// format converts date/time back into a string for display
```
these look like note to self comments and should be removed

okkhoy commented on src/main/java/clod/storage/Storage.java:
```diff
@@ -0,0 +1,181 @@
+import clod.operations.Deadline;
+import clod.exceptions.ClodException;
+import clod.ui.Interactions;
+import clod.operations.TimeManager; // Import TimeManager
```
redundant comment

okkhoy commented on src/main/java/clod/storage/Storage.java:
```diff
@@ -0,0 +1,181 @@
+            sb.append(deadline.getDescriptionWithoutIcons()).append(" | ").append(TimeManager.formatForStorage(deadline.getBy()));
+        } else if (task instanceof Event) {
+            Event event = (Event) task;
+            sb.append(event.getDescriptionWithoutIcons()).append(" | ").append(TimeManager.formatForStorage(event.getFrom())).append(" | ").append(TimeManager.formatForStorage(event.getTo())); // Use getDescriptionWithoutTime()
```
long line alert! please split to multiple lines

Yshinprograms replied:
Have fixed the coding standard issues, thanks for reviewing!

okkhoy reviewed (commented):
@Yshinprograms 
generally good code; there are some minor coding standard violations. please fix before the final deadline

Yshinprograms committed:
Fixed minor coding standard violations
